plugins/modules/ise_radius_integration_workflow_manager.py

```diff
@@ -794,122 +793,141 @@ def get_want_authentication_policy_server(self, auth_policy_server):
+                return self
+
+            if not 1 <= accounting_port <= 65535:
+                self.msg = "The 'accounting_port' should be from 1 to 65535."
+                self.status = "failed"
+                return self
+
+            auth_server.update({"accountingPort": accounting_port})
+        else:
+            auth_server.update({"accountingPort": auth_server_details.get("accountingPort")})
 
         retries = auth_policy_server.get("retries")
         if not retries:
-            retries = "3"
-
-        retries = str(retries)
-        if not retries.isdigit():
-            self.msg = "The 'retries' should contain only from 0-9."
-            self.status = "failed"
-            return self
+            if not auth_server_exists:
+                auth_server.update({"retries": "3"})
+            else:
+                auth_server.update({"retries": auth_server_details.get("retries")})
+        else:
+            retries = str(retries)
+            if not retries.isdigit():
+                self.msg = "The 'retries' should contain only from 0-9."
+                self.status = "failed"
+                return self
 
-        if not 1 <= int(retries) <= 3:
-            self.msg = "The 'retries' should be from 1 to 3."
-            self.status = "failed"
-            return self
+            if not 1 <= int(retries) <= 3:
+                self.msg = "The 'retries' should be from 1 to 3."
+                self.status = "failed"
+                return self
 
-        auth_server.update({"retries": retries})
+            auth_server.update({"retries": retries})
 
         timeout = auth_policy_server.get("timeout")
         if not timeout:
-            timeout = "4"
-
-        timeout = str(timeout)
-        if not timeout.isdigit():
-            self.msg = "The 'timeout' should contain only from 0-9."
-            self.status = "failed"
-            return self
+            if not auth_server_exists:
+                auth_server.update({"timeoutSeconds": "4"})
+            else:
+                auth_server.update({"timeoutSeconds": auth_server_details.get("timeoutSeconds")})
+        else:
+            timeout = str(timeout)
+            if not timeout.isdigit():
+                self.msg = "The 'timeout' should contain only from 0-9."
+                self.status = "failed"
+                return self
 
-        if not 2 <= int(timeout) <= 20:
-            self.msg = "The 'timeout' should be from 2 to 20."
-            self.status = "failed"
-            return self
+            if not 2 <= int(timeout) <= 20:
+                self.msg = "The 'timeout' should be from 2 to 20."
+                self.status = "failed"
+                return self
 
-        auth_server.update({"timeoutSeconds": timeout})
+            auth_server.update({"timeoutSeconds": timeout})
 
-        role = auth_policy_server.get("role")
-        if role:
-            auth_server.update({"role": role})
+        if not auth_server_exists:
```

Comment:
```
# Determine the role based on whether the auth server exists and if the role is specified
if not auth_server_exists:
    # Use the role from 'auth_policy_server' if available, otherwise default to "secondary"
    role = auth_policy_server.get("role", "secondary")
else:
    # Use the role from 'auth_server_details'
    role = auth_server_details.get("role")

auth_server.update({"role": role})
```

OR

```
if not auth_server_exists:
    role = auth_policy_server.get("role")
    # If role is not specified, set it to "secondary"
    if role is None:
        role = "secondary"
else:
    role = auth_server_details.get("role")

auth_server.update({"role": role})
```